main.js

```diff
@@ -1,104 +1,68 @@
 const articlesContainer = document.getElementById('articles-container');
+const pageBottom = document.getElementById('page-bottom');
 
 bindEvents();
 
 function bindEvents() {
+  window.addEventListener('load', () => {
+    const viewPortObserver = new IntersectionObserver(handleIntersections, {threshold: 0.5});
+    viewPortObserver.observe(pageBottom);
+  });
+  
   const submitButton = document.getElementById('submit');
 
   submitButton.addEventListener('click', event => {
     event.preventDefault();
+    sortSelect.value = 'relevance';
     submitNewSearch();
   });
   
   sortSelect.addEventListener('change', () => {
     toggleLoading();
-    resultsPage = 0;
     fetchArticles().then(() => {
+      submitNewSearch();
       toggleLoading();
-      displaySearchResults();
     });
   });
 
   filtersButton.addEventListener('click', event => {
     event.preventDefault();
     toggleFilterMenuVisibility();
   });
-
-  previousPageButton.addEventListener('click', () => {
-    toggleLoading();
-    resultsPage--;
-    fetchArticles().then(() => {
-      toggleLoading();
-      displaySearchResults();
-    });
-    scroll(0, 0);
-  });
-
-  nextPageButton.addEventListener('click', () => {
-    toggleLoading();
-    resultsPage++;
-    fetchArticles().then(() => {
-      toggleLoading();
-      displaySearchResults();
-    });
-    scroll(0, 0);
-  });
-}
-
-function configurePaginationButtons(pageArticles) {
-  const numArticlesOnPage = pageArticles.length;
-  
-  if (numArticlesOnPage < 10) {
-    nextPageButton.disabled = true;
-    nextPageButton.classList.add('disabled');
-  } else {
-    nextPageButton.disabled = false;
-    nextPageButton.classList.remove('disabled');
-  }
-
-  if (resultsPage === 0) {
-    previousPageButton.disabled = true;
-    previousPageButton.classList.add('disabled');
-  } else {
-    previousPageButton.disabled = false;
-    previousPageButton.classList.remove('disabled');
-  }
 }
 
 function displaySearchResults() {
-  const searchResultsDiv = document.getElementById('search-results-container');
-  const sortControls = document.getElementById('sort-by-container');
-
-  searchResultsDiv.style.display = 'none';
-  sortControls.style.display = 'none';
-
-  while (articlesContainer.firstChild) {
-    articlesContainer.removeChild(articlesContainer.firstChild);
-  }
-
   const totalHits = articles.response.meta.hits; 
   displayTotalHits(totalHits);
 
   if (totalHits > 0) {
+    // API limits pagination to 1000 articles (100 pages)
+    const totalScrollableHits = totalHits > 1000 ? 1000 : totalHits;
+    const totalScrollablePages = Math.floor(totalScrollableHits / 10);
     const currentPageArticles = articles.response.docs;
     
     currentPageArticles.forEach(article => {
       const articleHTML = getArticleHTML(article);
       articlesContainer.appendChild(articleHTML);
     });
 
+    if (resultsPage === totalScrollablePages) {
```

Comment:
check this out: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator